src/core_modules/capture-core/components/Breadcrumbs/EventBreadcrumb/hooks/useWorkingListLabel.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import { useMemo } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { useSelector } from 'react-redux';
+import { useApiMetadataQuery } from '../../../../utils/reactQueryHelpers';
+
+type Props = {
+    programId: string,
+}
+
+export const useWorkingListLabel = ({ programId }: Props) => {
+    const workingListTemplate = useSelector(({ workingListsTemplates }) => workingListsTemplates?.eventList);
+    const workingListProgramId = useSelector(({ workingListsContext }) => workingListsContext
+        ?.eventList
+        ?.programIdView,
+    );
+
+    const { selectedTemplateId, loading } = workingListTemplate ?? {};
+    const isDefaultTemplate = selectedTemplateId === `${programId}-default`;
+    const isSameProgram = workingListProgramId === programId;
+
+    const { data: eventFilterLabel, isLoading } = useApiMetadataQuery(
+        ['BreadCrumbs', 'workingListLabel', 'eventList', programId, selectedTemplateId],
+        {
+            resource: 'eventFilters',
+            id: selectedTemplateId,
+            params: {
+                fields: 'id,displayName',
+            },
+        },
+        {
+            enabled: !loading && !!selectedTemplateId && !isDefaultTemplate && isSameProgram,
+        },
+    );
```

Comment:
Same concern as mentioned here. If selectedTemplateId is stored in Redux, its name is also stored in Redux in the workingListsTemplates.eventList.templates array.